_source/_posts/2020-12-16-react-login.md

```diff
@@ -0,0 +1,599 @@
+
+```css
+.App header {
+  background-color: #282c34;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+  color: white;
+  padding: 0.5rem 1rem;
+}
+
+ul.menu {
+  list-style: none;
+}
+
+ul.menu li {
+  display: inline;
+  padding: 12px;
+}
+
+ul.menu a {
+  color: #ffffff;
+}
+
+.page {
+  text-align: center;
+}
+
+button {
+  background-color: #6200ee;
+  color: #ffffff;
+  border: none;
+  padding: 12px 16px;
+  text-transform: uppercase;
+  font-weight: bold;
+  cursor: pointer;
+}
+
+button:hover {
+  background-color: #3700b3;
+}
+
+.warning {
+  width: 800px;
+  margin: 24px auto;
+  border: 5px solid #ff0000;
+  border-radius: 15px;
+}
+
+.warning h1 {
+  margin-top: 0;
+  color: white;
+  font-size: 48px;
+  line-height: 96px;
+  background-color: #ff0000;
+  text-transform: uppercase;
+}
+
+.warning h2 {
+  font-size: 36px;
+  text-transform: uppercase;
+}
+```
+
+This completes the basic application. You can test it by running the following command.
+
+```bash
+npm start
+```
+
+This should build the JavaScript code and then automatically open your browser at `http://localhost:3000/`. You will see the home page and when you click the button, you will be taken to the restricted page. At this point, anybody could see the restricted page. No user authentication has been added so far.
+
+{% img blog/react-login/react-login-restricted-page.png alt:"Restricted page that should only be accessible to registered users" width:"800" %}{: .center-image }
+
+## Adding User Login with Redirection
+
+The simplest way to add authentication to the app is to use Okta's hosted login page. When users need to sign in, they are directed away from the site to a login form on the Okta servers. On success, the user is redirected back to the application. The authentication token is passed to the application and the validity is checked. This allows the application to open up restricted content that is hidden from unauthenticated users.
+
+This sort of authentication flow is ideally suited for single sign-on situations. For example, within a corporate network, a user logs in once through the central authentication provider. After that, the user has access to all applications within the corporate environment.
+
+To add authentication to the application, install the Okta libraries for React by running the command below.
+
+```bash
+npm install -E @okta/okta-react@4.1.0 @okta/okta-auth-js@4.4.0
+```
+
+The Okta React SDK has a peer dependency on Okta Auth JS, which provides most of the heavy lifting. In previous versions, this was included. Now you have to install both packages.
```

Comment:
In previous versions of what? The Okta React SDK?

Reply:
I'll add "of the Okta React SDK"